tests/js/spec/components/idBadge/teamBadge.spec.tsx

```diff
@@ -0,0 +1,45 @@
+import {act, mountWithTheme, screen} from 'sentry-test/reactTestingLibrary';
+
+import {TeamBadge} from 'sentry/components/idBadge/teamBadge';
+import TeamStore from 'sentry/stores/teamStore';
+
+describe('TeamBadge', function () {
+  beforeEach(() => {
+    TeamStore.init();
+  });
+
+  it('renders with Avatar and team name', async function () {
+    mountWithTheme(<TeamBadge team={TestStubs.Team()} />, {
+      context: TestStubs.routerContext(),
+    });
+    expect(screen.getByTestId('badge-styled-avatar')).toBeInTheDocument();
+    expect(screen.getByText(/#team-slug/)).toBeInTheDocument();
+  });
+
+  it('listens for avatar changes from TeamStore', async function () {
+    const team = TestStubs.Team();
```

Comment:
should we move this const to the very top and reuse in the tests?

Reply:
I dont think we should - that opens up to potential bugs in the future if the component has the ability of mutating the team and breaking other tests down the line. This way we make sure the tests run in isolation from each other